src/dawn/engine_dawn.cc

```diff
@@ -471,67 +608,253 @@ Result EngineDawn::DoClear(const ClearCommand* command) {
   return map.result;
 }
 
-Result EngineDawn::DoDrawRect(const DrawRectCommand*) {
+Result Helper::CreateRenderPipelineDescriptor(
+    const RenderPipelineInfo& render_pipeline,
+    const ::dawn::Device& device) {
   Result result;
 
-#if 0
-  // Add one more command to the command buffer builder, which is to
-  // copy the framebuffer texture to the framebuffer host-side buffer.
-  ::dawn::Buffer& fb_buffer = render_pipeline_info_.fb_buffer;
-  if (!fb_buffer) {
-    return Result(
-        "Dawn::DoDrawRect: Framebuffer was not created.  Did you run "
-        "any graphics pipeline commands?");
-  }
-
-  result = CreateCommandBufferBuilderIfNeeded();
+  auto* amber_format =
+      render_pipeline.pipeline->GetColorAttachments()[0].buffer->GetFormat();
+  if (!amber_format)
+    return Result("Color attachment 0 has no format!");
+  ::dawn::TextureFormat fb_format{};
+  result = GetDawnTextureFormat(*amber_format, &fb_format);
   if (!result.IsSuccess())
     return result;
 
-  {
-    const int x = 0, y = 0, z = 0, depth = 1, level = 0, slice = 0,
-              buffer_offset = 0;
+  ::dawn::TextureFormat depth_stencil_format{};
+  auto* depthBuffer = render_pipeline.pipeline->GetDepthBuffer().buffer;
+  if (depthBuffer) {
+    auto* amber_depth_stencil_format = depthBuffer->GetFormat();
+    if (!amber_depth_stencil_format)
+      return Result("The depth/stencil attachment has no format!");
+    result = GetDawnTextureFormat(*amber_depth_stencil_format,
+                                  &depth_stencil_format);
+    if (!result.IsSuccess())
+      return result;
+  } else {
+    depth_stencil_format = ::dawn::TextureFormat::D32FloatS8Uint;
+  }
 
-    ::dawn::TextureCopyView texture_copy_view = ::dawn::TextureCopyView();
-    texture_copy_view.texture = render_pipeline_info_.fb_texture;
-    texture_copy_view.level = level;
-    texture_copy_view.slice = slice;
-    texture_copy_view.origin = {x, y, z};
+  renderPipelineDescriptor.layout = MakeBasicPipelineLayout(device, nullptr);
+  renderPipelineDescriptor.primitiveTopology =
+      ::dawn::PrimitiveTopology::TriangleList;
+  renderPipelineDescriptor.sampleCount = 1;
+
+  // Set defaults for the vertex stage descriptor.
+  snprintf(entryPoint,
+           sizeof(render_pipeline.pipeline->GetShaders()[0].GetEntryPoint()),
+           "%s",
+           render_pipeline.pipeline->GetShaders()[0].GetEntryPoint().c_str());
+  cVertexStage.module = render_pipeline.vertex_shader;
+  cVertexStage.entryPoint = entryPoint;
+  renderPipelineDescriptor.vertexStage = &cVertexStage;
+
+  // Set defaults for the fragment stage descriptor.
+  cFragmentStage.module = render_pipeline.fragment_shader;
+  cFragmentStage.entryPoint = entryPoint;
```

Comment:
The fragment shader entry point name can be different.  You need to look it up again.  And I think do that by fragment shader type?